test/xcribe/request/validator_test.exs

```diff
@@ -0,0 +1,126 @@
+defmodule Xcribe.Request.ValidatorTest do
+  use ExUnit.Case, async: true
+
+  alias Xcribe.Request
+  alias Xcribe.Request.{Error, Validator}
+
+  describe "validate/1" do
+    test "return ok when request is valid" do
+      request = %Request{
+        action: "index",
+        controller: Elixir.Xcribe.UsersController,
+        description: "",
+        header_params: [{"authorization", "token"}],
+        params: %{},
+        path: "/users",
+        path_params: %{},
+        query_params: %{},
+        request_body: %{},
+        resource: "users",
+        resource_group: :api,
+        resp_body: "[{\"id\":1,\"name\":\"user 1\"},{\"id\":2,\"name\":\"user 2\"}]",
+        resp_headers: [
+          {"content-type", "application/json; charset=utf-8"},
+          {"cache-control", "max-age=0, private, must-revalidate"}
+        ],
+        status_code: 200,
+        verb: "get"
+      }
+
+      assert Validator.validate(request) == {:ok, request}
+    end
+
+    test "return an error when request has a struct in path_params" do
+      meta = %{metadata: "value"}
+      request = %Request{path_params: ~D[2021-02-15], __meta__: meta}
+
+      assert Validator.validate(request) ==
+               {:error,
+                %Error{
+                  type: :validation,
+                  message:
+                    "The Plug.Conn params must be valid HTTP params. A struct Elixir.Date was found!",
```

Comment:
```suggestion
                    "The Plug.Conn params must be valid HTTP params. A struct Date was found!",
```

@Danielwsx64 Do you think would be nicer for the user to remove the Elixir prefix?

Reply:
Done :v: